lib/string-checker.js

```diff
@@ -83,17 +83,6 @@ StringChecker.prototype = {
      * @throws Error on invalid source code.
      */
     _parse: function(str) {
-        var hashbang = str.indexOf('#!') === 0;
-        var tree;
-
-        // Convert bin annotation to a comment
-        if (hashbang) {
-            str = '//' + str.substr(2);
-        }
-
-        // Strip special case code like iOS instrumentation imports: `#import 'abc.js';`
-        str = str.replace(/^#!?[^\n]+\n/gm, '');
-
         var esprimaOptions = {
```

Comment:
I suppose this, and the best of the function body here, should move into in `JsFile.parse` too. You're doing one too many copies of the options.